src/melee.cpp

```diff
@@ -251,7 +251,7 @@ void player::roll_all_damage( bool crit, damage_instance &di, bool average, cons
 static void melee_train( player &p, int lo, int hi ) {
     p.practice( skill_melee, ceil( rng( lo, hi ) / 2.0 ) );
 
-    if( p.unarmed_attack() ) {
+    if( !p.is_armed() ) {
```

Comment:
Is this special case needed?
It probably doesn't hurt, but looks totally redundant, considering that the check below now covers all sorts of unarmed.